lightning/src/routing/router.rs

```diff
@@ -1863,14 +1886,17 @@ where L::Target: Logger {
 						.saturating_add(1);
 
 					// Searching for a direct channel between last checked hop and first_hop_targets
-					if let Some(first_channels) = first_hop_targets.get(&NodeId::from_pubkey(&prev_hop_id)) {
+					let hint_candidate_contribution_msat = cmp::min(path_value_msat,
```

Comment:
I think we need this because otherwise `add_entry` assumes that the next hop (aka the route hint hop) can carry the full payment value, since `available_value_contribution_msat` only factors in the current candidate's capacity, not the next hop candidate's. `direct_channel_to_hints_with_max_htlc` fails without this.